src/main/java/org/prebid/server/cache/CacheService.java

```diff
@@ -219,6 +242,39 @@ private static String createUuid(CacheObject cacheObject) {
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Creates a map with bids as a key and {@link CacheIdInfo} as a value from obtained UUIDs.
+     */
+    private static <T> Map<T, CacheIdInfo> toResultMap(List<T> bids, List<T> videoBids, List<String> uuids) {
+        final Map<T, CacheIdInfo> result = new HashMap<>(uuids.size());
+
+        for (int i = 0; i < bids.size(); i++) {
+            final T bid = bids.get(i);
+            if (!result.containsKey(bid)) {
+                final String bidUuid = uuids.get(i);
+
+                final int indexOf = videoBids.indexOf(bid);
+                final String videoBidUuid = indexOf != -1 ? uuids.get(i + indexOf) : null;
+
+                result.put(bid, CacheIdInfo.of(bidUuid, videoBidUuid));
+            }
+        }
+
+        for (int i = 0; i < videoBids.size(); i++) {
```

Comment:
Do I understand correct that this loop has sense only when bids is empty collection? If not, all video bids will be already in result.